Ignore the available volume when calling DetachDisk

FengyunPan opened the pull request:
Fix #50207
If user detachs the volume by nova in openstack env, volume becomes
available. If nova instance is been deleted, nova will detach it
automatically and become available. So the "available" is fine since that means the
volume is detached from instance already.

**Release note**:
```release-note
NONE
```

dims commented:
/ok-to-test

FengyunPan commented:
/test pull-kubernetes-e2e-kops-aws

kars7e commented on pkg/cloudprovider/providers/openstack/openstack_volumes.go:
```diff
@@ -258,10 +254,10 @@ func (os *OpenStack) DetachDisk(instanceID, volumeID string) error {
-		errmsg := fmt.Sprintf("can not detach volume %s, its status is %s.", volume.Name, volume.Status)
-		glog.Errorf(errmsg)
-		return errors.New(errmsg)
+	if volume.Status == VolumeAvailableStatus {
```
What about statuses other than in-use or available? Shouldn't we return an error if volume is in error state?

FengyunPan replied:
Yeah, I should restore the original codes, thank for your comment, I will update it soon.

Commit:
Ignore the available volume when calling DetachDisk

If use detachs the volume by nova in openstack env, volume becomes
available. If nova instance is been deleted, nova will detach it
automatically. So the "available" is fine since that means the
volume is detached from instance already.

FengyunPan commented:
/retest

FengyunPan commented:
/test pull-kubernetes-e2e-gce-etcd3

dims commented:
/approve

dims commented:
/lgtm
